libraries/engage/product/components/Swatch/SwatchContent.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import noop from 'lodash/noop';
+import cls from 'classnames';
+import { PRODUCT_SWATCH } from '@shopgate/pwa-common-commerce/product/constants/Portals';
+import { SurroundPortals } from '../../../components';
+import SwatchColor from './SwatchColor';
+import SwatchTexture from './SwatchTexture';
+import { swatchClass, itemClass, itemSelectedClass } from './style';
+
+/**
+ * The swatch content component.
+ * @param {Object} props The component props.
+ * @returns {JSX}
+ */
+const SwatchContent = ({ swatch, classNames, onClick }) => {
+  if (!swatch) {
+    return null;
+  }
+
+  return (
+    <SurroundPortals portalName={PRODUCT_SWATCH} portalProps={{ swatch }}>
+      <ul className={cls(swatchClass, {
+          [classNames.swatch]: !!classNames.swatch,
+        })}
+      >
+        {swatch.values.map((value) => {
+          if (value.swatch.color) {
+            return (
+              <SwatchColor
+                key={value.id}
+                valueId={value.id}
+                color={value.swatch.color}
```

Comment:
The only difference of these two cases is whether it has a color or an image url. You should do the conditional check inline, to avoid this code duplication:
```
color={value.swatch.color || null}
imageUrl={value.swatch.imageUrl || null}
```